hysds/user_rules_dataset.py

```diff
@@ -89,6 +89,13 @@ def update_query(_id, system_version, rule):
     return final_query
 
 
+@backoff.on_exception(
+    backoff.expo, Exception, max_tries=backoff_max_tries, max_value=backoff_max_value
+)
+def search_es(index, body):
+    return grq_es.es.search(index=index, body=body)
```

Comment:
another suggestion is to return only the relevant fields needed, that can potentially return the data faster from ES
it looks like user rules doesnt need any of the fields returned so its safe to return an empty `_source`

https://elasticsearch-py.readthedocs.io/en/v7.14.0/api.html#elasticsearch.Elasticsearch.search
> `_source` – True or false to return the _source field or not, or a list of fields to return

Reply:
Good catch. I think that makes sense in terms of user rule evaluation. @mkarim2017 @pymonger what do you think?

Reply:
> another suggestion is to return only the relevant fields needed it can potentially return the data faster from ES it looks like user rules doesnt need any of the fields returned so its safe to return an empty `_source`
> 
> https://elasticsearch-py.readthedocs.io/en/v7.14.0/api.html#elasticsearch.Elasticsearch.search
> 
> > `_source` – True or false to return the _source field or not, or a list of fields to return

@DustinKLo @mcayanan @gmanipon 
 I see it uses the whole doc in doc_res = result["hits"]["hits"][0]. Also if we want to use this function as generic like other functions can also can call it, we dont know what fields to return. May be it is better to have the query updated to only look for those fields that needed?

Reply:
That's right. the `dataset_jpath` feature allows the HySDS-IO param configurations to path down to any field in the body of the triggering dataset.

Reply:
@DustinKLo: you make a great point though with optimizing our ES queries especially for trigger rule evaluation. The return of the entire doc can be a very heavy operation especially in some of the NISAR dataset cases. It would be best to track this as a separate ticket though so it stays on our radar.